code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,316 @@
+			return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/attack_hand(mob/user as mob)
+	return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(air_group)
+		return state
+	if(istype(mover, /obj/effect/beam))
+		return !opacity
+	return !density
+
+/obj/structure/inflatable/door/CanAtmosPass(turf/T)
+	return !density
+
+/obj/structure/inflatable/door/proc/TryToSwitchState(atom/user)
+	if(isSwitchingStates)
+		return
+	if(ismob(user))
+		var/mob/M = user
+		if(M.client)
+			if(iscarbon(M))
+				var/mob/living/carbon/C = M
+				if(!C.handcuffed)
+					SwitchState()
+			else
+				SwitchState()
+	else if(istype(user, /obj/mecha))
+		SwitchState()
+
+/obj/structure/inflatable/door/proc/SwitchState()
+	if(state)
+		Close()
+	else
+		Open()
+	air_update_turf(1)
+
+/obj/structure/inflatable/door/proc/Open()
+	isSwitchingStates = 1
+	//playsound(loc, 'sound/effects/stonedoor_openclose.ogg', 100, 1)
+	flick("door_opening",src)
+	sleep(10)
+	density = 0
+	opacity = 0
+	state = 1
+	update_icon()
+	isSwitchingStates = 0
+
+/obj/structure/inflatable/door/proc/Close()
+	isSwitchingStates = 1
+	//playsound(loc, 'sound/effects/stonedoor_openclose.ogg', 100, 1)
+	flick("door_closing",src)
+	sleep(10)
+	density = 1
+	opacity = 0
+	state = 0
+	update_icon()
+	isSwitchingStates = 0
+
+/obj/structure/inflatable/door/update_icon()
+	if(state)
+		icon_state = "door_open"
+	else
+		icon_state = "door_closed"
+
+/obj/structure/inflatable/door/deflate(var/violent=0)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("<span class='danger'>[src] rapidly deflates!</span>")
+		var/obj/item/inflatable/door/torn/R = new /obj/item/inflatable/door/torn(loc)
+		src.transfer_fingerprints_to(R)
+		qdel(src)
+	else
+		visible_message("<span class='danger'>[src] slowly deflates.</span>")
+		spawn(50)
+			var/obj/item/inflatable/door/R = new /obj/item/inflatable/door(loc)
+			src.transfer_fingerprints_to(R)
+			qdel(src)
+	air_update_turf(1)
+
+/obj/item/inflatable/torn
+	name = "torn inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation. It is too torn to be usable."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_wall_torn"
+
+/obj/item/inflatable/torn/attack_self(mob/user)
+	user << "<span class='warning'>The inflatable wall is too torn to be inflated, fix it with something!</span>"
+	add_fingerprint(user)
+
+/obj/item/inflatable/torn/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack/ducttape))
+		var/obj/item/stack/ducttape/T = I
+		if(T.amount < 3)
+			user << "<span class='danger'>There is not enough tape!</span>"
+			return
+		user << "<span class='notice'>You begin fixing the [src]!</span>"
+		playsound(user, 'sound/items/ducttape1.ogg', 50, 1)
+		if(do_mob(user, src, 20))
+			user << "<span class='notice'>You fix the [src] using the ducttape!</span>"
+			T.amount = T.amount - 3
```

Comment:
There is an exact proc to edit the amount,otherwise you could get a duct tape item with 0 quantity. It's called use(used)